module/os/freebsd/zfs/zvol_os.c

```diff
@@ -1305,6 +1361,8 @@ zvol_os_free(zvol_state_t *zv)
 
 		if (dev != NULL) {
 			ASSERT3P(dev->si_drv2, ==, NULL);
+			knlist_clear(&zsd->zsd_selinfo.si_note, 0);
+			knlist_destroy(&zsd->zsd_selinfo.si_note);
 			destroy_dev(dev);
```

Comment:
I suspect that knlist_clear()/knlist_destroy() should go after destroy_dev().  At least that is what I see in few other places in kernel.

Reply:
Hmm, I looked through the code a bit to understand why that is. One of the other patterns I noticed were variations of:

```
destroy_dev()
seldrain()
knlist_clear()
knlist_destroy()
```

I didn't come up with a confident conclusion - my best guess is that devices with `d_poll()` handlers should be destroyed before draining `selinfo`. Since the knote list is attached to `selinfo`, it would make sense to clear/destroy the knlist after the device is destroyed.

I've updated this chunk of the pull request to be consistent with rest of the kernel.